### ISSUE TITLE
Date and time not being set for spdlog::sinks::basic_file_sink_mt

### ISSUE THREAD

**Issue by williamleong** [gabime/spdlog#2932]
Does anyone experience an issue where the logged data doesn't show the date and time? This is happening to me in v1.10.0 (76fb40d9).

For example, with :
````
logger = spdlog::create_async<spdlog::sinks::basic_file_sink_mt>(filename, fullPath);
````
I am only getting data like:
````
[1900-01-00 00:00:00.059] [filename] [info] 4 800.00 0.00                                                                                                                        
[1900-01-00 00:00:00.159] [filename] [info] 1 400.00 0.00                                                                                                                        
[1900-01-00 00:00:00.159] [filename] [info] 2 800.00 0.00                                                                                                                        
[1900-01-00 00:00:00.159] [filename] [info] 3 400.00 0.00                                                                                                                        
[1900-01-00 00:00:00.159] [filename] [info] 4 800.00 0.00                                                                                                                        
[1900-01-00 00:00:00.259] [filename] [info] 1 400.00 0.00                                                                                                                        
[1900-01-00 00:00:00.259] [filename] [info] 2 800.00 0.00                                                                                                                        
````

If I roll back to v1.9.2 (eb322062), the problem goes away and I get the expected date and time values:
````
[2023-11-07 16:37:13.509] [filename] [info] 2 308.13 0.13
[2023-11-07 16:37:13.509] [filename] [info] 3 281.95 0.20
[2023-11-07 16:37:13.509] [filename] [info] 4 627.39 0.14
[2023-11-07 16:37:13.609] [filename] [info] 0 256.72 0.28
[2023-11-07 16:37:13.609] [filename] [info] 2 307.62 0.12
[2023-11-07 16:37:13.609] [filename] [info] 3 279.90 0.18
[2023-11-07 16:37:13.609] [filename] [info] 4 625.38 0.12
````

From what I can tell, `need_localtime_` in pattern_formatter-inl.h is false and therefore the following code doesn't run to set `cached_tm_`:
````
    if (need_localtime_)
    {
        const auto secs = std::chrono::duration_cast<std::chrono::seconds>(msg.time.time_since_epoch());
        if (secs != last_log_secs_)
        {
            cached_tm_ = get_time_(msg);
            last_log_secs_ = secs;
        }
    }
````

**Comment by tt4g** [comment]
Depending on the constructor, `need_localtime_` can be `false`.

https://github.com/gabime/spdlog/blob/ac55e60488032b9acde8940a5de099541c4515da/include/spdlog/pattern_formatter-inl.h#L872-L895

Later, `need_localtime_` may be changed to `true` in `handle_flag_()` (called through `compile_pattern_()`).

https://github.com/gabime/spdlog/blob/ac55e60488032b9acde8940a5de099541c4515da/include/spdlog/pattern_formatter-inl.h#L945

Are you changing the pattern?

**Comment by williamleong** [comment]
Thanks for checking. I'm using the default pattern, so I don't think that's it.

**Comment by tt4g** [comment]
It is possible that the ABI has changed between the two versions and the referenced memory is incorrect.
Has the project been re-built after the spdlog upgrade?

**Comment by williamleong** [comment]
I tried rebuilding and I found out that the problem is due to some strange interactions when a dependency in my project has fmt as a dependency and somehow this results in referencing the system installed fmt. It is possibly related to #2894 as the versions of spdlog and fmt are the same as what I am using, but the solution is not applicable for me as I do not want to install the libraries or use external fmt.

Due to time constraints, I can't investiate further to find out the actual cause.

Thanks @tt4g, for your suggestions.
